crates/cargo-winrt/src/main.rs

```diff
@@ -438,70 +438,30 @@ impl DependencyDescriptor {
     }
 }
 
-fn try_download(url: String, recursion_amount: u8) -> anyhow::Result<Vec<u8>> {
-    if recursion_amount == 0 {
-        bail!(Error::DownloadError(
-            anyhow::anyhow!("Too many redirects").into(),
-        ));
-    }
+fn try_download(url: String) -> anyhow::Result<Vec<u8>> {
     let mut handle = Easy::new();
     handle
         .url(&url)
         .map_err(|e| Error::DownloadError(e.into()))?;
-    let status = &mut None;
+    // Instruct curl to follow redirections
+    handle.follow_location(true)?;
+    // Optionally set the max number of redirects
+    handle.max_redirections(5)?;
+
+    let mut bytes = Vec::new();
     {
         let mut transfer = handle.transfer();
-        transfer
-            .header_function(|header| {
-                let header = std::str::from_utf8(header).unwrap();
-                if header.starts_with("HTTP/1.1 ") {
-                    let n = &header[9..12];
-                    *status = Some(n.parse::<u16>().expect("Should be number"));
-                }
-                true
-            })
-            .unwrap();
+        transfer.write_function(|d| {
+            bytes.extend_from_slice(d);
+            Ok(d.len())
+        })?;
         print_verbose_status!("Requesting", &url);
         transfer.perform()?;
```

Comment:
Probably here, too, though you should probably double-check with @rylev why the original code didn't. Maybe there's more to it.

Reply:
We should map this to a DownloadError as well. Just add `.map_err(|e| Error::DownloadError(e.into()))` to all of these.